crates/ra_hir/src/db.rs

```diff
@@ -36,21 +36,9 @@ pub trait HirDatabase: SourceDatabase + AsRef<HirInterner> {
     #[salsa::invoke(crate::adt::EnumData::enum_data_query)]
     fn enum_data(&self, e: Enum) -> Arc<EnumData>;
```

Comment:
The defenitions of the queries (ie `adt::EnumData::enum_data_query`) should be changed as well to take `db: &impl PersistentHirDatabase` query instead of `HirDatabase`.

Reply:
yep, I'm clean up those kind of `impl trait` mismatching locally.